Collab: Sync breakpoints

RemcoSmitsDev opened the pull request:
https://github.com/user-attachments/assets/d546379a-31db-4afa-b177-b4993a99e5bd

----

### Todo
- [x] Sync breakpoints on breakpoint toggle
- [x] Sync breakpoints on project share/join
  - [x] Sync breakpoints when someone connects after you setting your first breakpoint

RemcoSmitsDev committed:
Sync breakpoints on toggle to other client

RemcoSmitsDev committed:
WIP Seperate local/remote in dap store

RemcoSmitsDev committed:
WIP initial breakpoints sync

Co-Authored-By: Anthony Eid <56899983+Anthony-Eid@users.noreply.github.com>

Anthony-Eid committed:
Get zed to compile

RemcoSmitsDev committed:
Don't remove dap data when you unshare

RemcoSmitsDev committed:
Add breakpoints table migration

RemcoSmitsDev committed:
Update collab db when changing breakpoints

RemcoSmitsDev committed:
Store breakpoints inside collab db when you change them

RemcoSmitsDev committed:
Clean up

Anthony-Eid commented on crates/collab/src/db/queries/projects.rs:
```diff
@@ -476,6 +477,60 @@ impl Database {
+                .ok_or_else(|| anyhow!("no such project"))?;
+
+            // remove all existing breakpoints
+            breakpoints::Entity::delete_many()
```
I think we should be deleting all breakpoints associated with the project_id and project_path instead of just the project_id. 

If I'm understanding this correctly, update only contains a list of breakpoints per one file. So when the breakpoints from update are inserted into the db, breakpoints from other file's won't be updated correctly and would be deleted.

RemcoSmitsDev replied:
Oops, yeah correct!

Anthony-Eid committed:
Fix incorrect clearing of breakpoints during collab sync

Anthony-Eid committed:
Get breakpoints to sync correctly on project join

We now send SynchronizedBreakpoints within the JoinProjectResponse
and use those breakpoints to initialize a remote DapStore.

Anthony-Eid commented:
I think this is ready to be merged!

RemcoSmitsDev commented on crates/project/src/dap_store.rs:
```diff
@@ -145,8 +145,24 @@ impl DapStore {
+        breakpoints: Option<Vec<proto::SynchronizeBreakpoints>>,
         _: &mut ModelContext<Self>,
     ) -> Self {
+        let breakpoints = match breakpoints {
```
I'm not sure if I like this approach. I feel like we have to add this somewhere else, this feels like a hack. If this is how they do it in other cases, i'm fine with it! Let's merge it

Anthony-Eid committed:
Set breakpoints from proto method
